inference-engine/tests/functional/shared_test_classes/include/shared_test_classes/base/layer_test_utils.hpp

```diff
@@ -105,6 +111,36 @@ class LayerTestsCommon : public CommonTestUtils::TestsCommon {
             double diff = static_cast<float>(absoluteDifference) / max;
             if (max == 0 || (diff > static_cast<float>(threshold)) ||
                 (std::isnan(static_cast<float>(res)) ^ std::isnan(static_cast<float>(ref)))) {
+                IE_THROW() << "Relative comparison of values expected: " << std::to_string(ref) << " and actual: " << std::to_string(res)
+                           << " at index " << i << " with relative threshold " << threshold
+                           << " failed";
+            }
+        }
+    }
+
+    template<class T_IE, class T_NGRAPH>
+    static void Compare(const T_NGRAPH *expected, const T_IE *actual, std::size_t size, float threshold, float absThreshold) {
+        if ((threshold < 0) && (absThreshold < 0)) {
+            IE_THROW() << "Both relative threshold and absolute threshold aren't set properly";
+        }
+        for (std::size_t i = 0; i < size; ++i) {
+            const T_NGRAPH &ref = expected[i];
+            const auto &res = actual[i];
+            const auto absoluteDifference = CommonTestUtils::ie_abs(res - ref);
+            if ((absoluteDifference > absThreshold) && (absThreshold >= 0)) {
```

Comment:
It is better not to throw on the first fail but collect statistics on the whole tensor and if there is any fail provide the mean and max absolute and relative difference for the whole tensor and the number of failed elements.
This approach allows us to understand whether it is a minor accuracy issue that can be fixed with threshold tweek in tests or it is a real accuracy bug.

Reply:
Understood your requirement and it is reasonable. Do you want to get those statistics when any one of tensor elements is bigger than threshold? Or mean is bigger than threshold?

Reply:
Below output prepared when any single element is bigger than threshold.

/home/intel-ava14/disk2/openvino_fork/openvino/inference-engine/tests/functional/shared_test_classes/include/shared_test_classes/base/layer_test_utils.hpp:180 Relative comparison diff tensor mean: 0.00110134, max: 0.967262, # failure(1/882), Absolute comparison diff tensor mean: 2.01673e-06, max: 9.53674e-06, # failure(0/882)